pkg/schemacompat/schemacompat_test.go

```diff
@@ -0,0 +1,217 @@
+package schemacompat
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestSuccessNewHasMoreProperties(t *testing.T) {
```

Comment:
These tests seem like a good opportunity for a [table-driven test](https://github.com/golang/go/wiki/TableDrivenTests).

The case struct looks like it would have `existing` and `new`, and a `want` result and (possibly `nil`) `wantErr` field, and the test body would check that `lcd` matches `c.want` and `err` matches `c.wantErr`.

Reply:
+1, but I'd rather let this for a next PR, to avoid postponing the merge of the whole API Negotiation work.